skops/card/_model_card.py

```diff
@@ -1,61 +1,120 @@
 import os
 import re
+import shutil
 
-from modelcards import ModelCard
+from modelcards import CardData, ModelCard
 from sklearn.utils import estimator_html_repr
 
 import skops
 
 
-def _extract_estimator_config(model):
-    """Extracts estimator configuration and renders them into a vertical table.
-
-    Parameters
-    ----------
-        model (estimator): scikit-learn pipeline or model.
-
-    Returns
-    -------
-    str:
-        Markdown table of hyperparameters.
-    """
-    hyperparameter_dict = model.get_params(deep=True)
-    table = "| Hyperparameters | Value |\n| :-- | :-- |\n"
-    for hyperparameter, value in hyperparameter_dict.items():
-        table += f"| {hyperparameter} | {value} |\n"
-    return table
-
-
-def create_model_card(
-    model,
-    card_data,
-    **card_kwargs,
-):
-    """Creates a model card for the model and saves it to the target directory.
-
-    Parameters:
-    ----------
-    model: estimator
-        scikit-learn compatible estimator.
-    card_data: CardData
-        CardData object.
-    card_kwargs:
-        Card kwargs are information you can pass to fill in the sections of the
-        card template, e.g. model_description, citation_bibtex, get_started_code.
-    """
-    ROOT = skops.__path__
-    model_plot = re.sub(r"\n\s+", "", str(estimator_html_repr(model)))
-    hyperparameter_table = _extract_estimator_config(model)
-    card_data.library_name = "sklearn"
-    template_path = card_kwargs.get("template_path")
-    if template_path is None:
-        template_path = os.path.join(f"{ROOT[0]}", "card", "default_template.md")
-    card_kwargs["template_path"] = template_path
-    card = ModelCard.from_template(
-        card_data=card_data,
-        hyperparameter_table=hyperparameter_table,
-        model_plot=model_plot,
-        **card_kwargs,
-    )
-
-    return card
+class Card:
+    def __init__(self, model):
+        self.model = model
+        self.hyperparameter_table = self._extract_estimator_config()
+        # the spaces in the pipeline breaks markdown, so we replace them
+        self.model_plot = re.sub(r"\n\s+", "", str(estimator_html_repr(model)))
+        self.template_sections = {}
+        self.figure_paths = {}
+
+    def add(self, **kwargs):
+        """Takes values to fill model card template.
```

Comment:
We need to add a `model_card.rst` under `docs/modules` and explain what the available arguments here are, and link to that one from here.

See [here](https://github.com/scikit-learn/scikit-learn/blob/2e49193e3fd9c12a710ecae12f6330c878de0540/sklearn/linear_model/_quantile.py#L29) on how to link to a user guide, and look for `.. _quantile_regression:` in [this file](https://raw.githubusercontent.com/scikit-learn/scikit-learn/main/doc/modules/linear_model.rst) to see an example of a user guide.

If you want to have a separate PR to add the user guide and then add links here, I'm fine with that.

Reply:
@merveenoyan has already opened the issue: https://github.com/skops-dev/skops/issues/50, but I think she forgot to put the link here.